File: kotlin/hydroponics-host/src/main/kotlin/io/greenglass/hydroponics.application/models/installation/InstallationViewModel.kt
package io.greenglass.hydroponics.application.models.installation

import org.neo4j.graphdb.Node

class InstallationViewModel(
    val installationId : String,
    val instanceName : String,
    val implementationId : String,
    val implementationName : String,
    val systemId : String,
    val systemName : String,
) {
    constructor(instNode : Node, implNode : Node, sysNode : Node) : this(
        instNode.getProperty("installationId") as String,
        instNode.getProperty("name") as String,
        implNode.getProperty("implementationId") as String,
        implNode.getProperty("name") as String,
        sysNode.getProperty("systemId") as String,
        sysNode.getProperty("name") as String,
    )
}

File: flutter/hydroponics_framework/lib/datamodels/installation/installation_view_model.dart
class InstallationViewModel {
  late String installationId;
  late String instanceName;
  late String implementationId;
  late String implementationName;
  late String systemId;
  late String systemName;

  InstallationViewModel(
      this.installationId,
      this.instanceName,
      this.implementationId,
      this.implementationName,
      this.systemId,
      this.systemName,
      );

  InstallationViewModel.fromJson(Map<String, dynamic> json) {
    installationId = json["installationId"];
    instanceName = json["instanceName"];
    implementationId = json["implementationId"];
    implementationName = json["implementationName"];
    systemId = json["systemId"];
    systemName = json["systemName"];
  }
}
